websockify/websocket.py

```diff
@@ -164,21 +169,34 @@ def encode_hybi(buf, opcode, base64=False):
             0x9 - ping
             0xA - pong
         """
-        if base64:
-            buf = b64encode(buf)
-
         b1 = 0x80 | (opcode & 0x0f) # FIN + opcode
-        payload_len = len(buf)
         if payload_len <= 125:
-            header = pack('>BB', b1, payload_len)
+            return pack('>BB', b1, payload_len)
         elif payload_len > 125 and payload_len < 65536:
-            header = pack('>BBH', b1, 126, payload_len)
-        elif payload_len >= 65536:
-            header = pack('>BBQ', b1, 127, payload_len)
+            return pack('>BBH', b1, 126, payload_len)
+        else:
+            assert payload_len >= 65536
+            return pack('>BBQ', b1, 127, payload_len)
+
+    @staticmethod
+    def encode_hybi(buf, opcode, base64=False):
+        """ Encode a HyBi style WebSocket frame. """
+        if base64:
+            buf = b64encode(buf)
+        header = WebSocketRequestHandler.encode_hybi_header(len(buf), opcode)
+        return (header, buf)
 
-        #self.msg("Encoded: %s", repr(header + buf))
+    def send_hybi(self, buf, opcode, base64=False, record=None):
+        """ Send a HyBi style WebSocket frame. """
+        header, buf = self.encode_hybi(buf, opcode, base64)
+        if record:
+            record.write("%s,\n" % repr("{%s{" % tdelta + encbufs[1]))
+        if len(buf)<=4096:
+            self.request.send(header+buf)
```

Comment:
because socket.send isn't guaranteed to have sent all the requested data, and it returns however much it sends.  Previously, the related methods here would return the result of self.request.send, and we would check to make sure the entirety of the buffer was sent.  If not, we'd remember the rest, and send it later.  This patch removes that functionality entirely (it can be seen here in the fact that we don't do anything with the result of `self.request.send`).